apps/pages/middleware.py

```diff
@@ -11,7 +11,10 @@ def process_response(self, request, response):
         if response.status_code != 404:
             return response  # No need to check for a flatpage for non-404 responses.
         try:
-            return PageDetailView.as_view()(request, url=request.path_info).render()
+            try:
```

Comment:
Slightly unsure on this.

Maybe something along the lines of:

```
page_response = PageDetailView.as_view()(request, url=request.path_info)

if page_response.return_code_or_whatever_it_is == not_allowed_response_code:
    return original_response

return page_response.render()
```

However it does work, so feel free to :shipit: and move onto the next bit.